spec/controllers/equipment_models_controller_spec.rb

```diff
@@ -1,421 +1,280 @@
         before do
-          put :update, id: model, equipment_model:
-          FactoryGirl.attributes_for(:equipment_model, name: 'Mod')
-        end
-        it { is_expected.to set_flash }
-        it 'sets @equipment_model to selected model' do
-          expect(assigns(:equipment_model)).to eq(model)
-        end
-        it 'updates attributes' do
-          model.reload
-          expect(model.name).to eq('Mod')
+          allow(EquipmentModel).to receive(:find).with(model.id.to_s)
+            .and_return(model)
+          allow(model).to receive(:destroy)
+          put :deactivate, id: model.id, **opts
         end
+        it { is_expected.to set_flash[flash_type] }
         it { is_expected.to redirect_to(model) }
+        it 'does not deactivate model' do
+          expect(model).not_to have_received(:destroy)
+        end
       end
+      it_behaves_like 'not confirmed', :error
+      it_behaves_like 'not confirmed', :notice, deactivation_cancelled: true
 
-      context 'without valid attributes' do
+      context 'confirmed' do
+        let!(:model) { mock_eq_model(traits: [:findable]) }
         before do
-          put :update, id: model, equipment_model:
-          FactoryGirl.attributes_for(:equipment_model, name: nil)
+          allow(EquipmentModel).to receive(:find).with(model.id.to_s)
+            .and_return(model)
+          request.env['HTTP_REFERER'] = 'where_i_came_from'
+          put :deactivate, id: model.id, deactivation_confirmed: true
         end
-        it { is_expected.not_to set_flash }
-        it 'should not update attributes' do
-          model.reload
-          expect(model.name).not_to be_nil
+        it { is_expected.to set_flash[:notice] }
+        it { is_expected.to redirect_to('where_i_came_from') }
+        it 'deactivates model' do
+          expect(model).to have_received(:destroy)
         end
-        it { is_expected.to render_template(:edit) }
       end
-      it 'calls delete_files'
+
+      context 'with reservations' do
+        it "archives the model's reservations on deactivation" do
+          model = mock_eq_model(traits: [:findable])
+          # FIXME: replace with mock method
+          res = instance_spy('reservation', equipment_model: model)
+          # stub out scope chain
+          allow(Reservation).to receive(:for_eq_model).and_return(Reservation)
+          allow(Reservation).to receive(:finalized).and_return([res])
+          expect(res).to receive(:archive).and_return(res)
+          expect(res).to receive(:save).with(validate: false)
+          request.env['HTTP_REFERER'] = 'where_i_came_from'
+          put :deactivate, id: model.id, deactivation_confirmed: true
+        end
+      end
+    end
+    context 'not admin' do
+      before do
+        mock_user_sign_in
+        put :deactivate, id: 1
+      end
+      it_behaves_like 'redirected request'
+    end
+  end
+
+  describe 'GET show' do
+    # the current controller method is too complex to be tested
```

Comment:
Yea, duly noted. We might want to look into refactoring this into a set of decorators or other objects that can be tested separately so that it isn't all the responsibility of a single 30-line method. The fact that we can't write nice specs for it as-is is definitely a smell by itself.